### PR TITLE
Fixes #1311, updated the public api and some improvement on UI

### LINKED ISSUE #1311: Documentation of public API (JSON Model Request/Response, Examples, ...)
Currently, if a user wants to try out the public REST - API, she has no idea what use cases the REST - API fullfills. Also there is no proper docuemtation how the Request/Response JSON looks like.

Recommended type of documentation:

    Richer documentation in Swagger
    Chapter on REST- API of Repository Readme.md.

For reference, take a look at the Eclipse Ditto Project Swagger Documentation.

### PR DESCRIPTION
**Pull request by vlvanchin**
Signed-off-by: vlvanchin <vanchinathan.lokanathan@bosch-si.com>

### CODE REVIEW

**Comment by vlvanchin** [comment]
Dear Alex,

Please trigger the build again, as the build got terminated due to "No output has been received in the last 10m0s". 

Thank you

vanchin

**Comment by aedelmann** [comment]
Hi @vlvanchin 
There seems to be a problem in the Arduino Generator Integration Tests, see travis log.

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java [hunk 1]
```diff
@@ -60,7 +60,14 @@
 
   private final Logger LOGGER = LoggerFactory.getLogger(getClass());
 
-  @ApiOperation(value = "Upload a file to be attached to a model")
+  @ApiOperation(value = "Upload a file to be attached to a model",
+		  notes = "This method is used to get 'UPLOAD' a single file attached to the specific modelId."
```
Comment (erlemantos) [inline repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java]: This method is used to upload

Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java]: Updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java [hunk 2]
```diff
@@ -60,7 +60,14 @@
...
+			  		+ "<br/>"
+			  		+ "<pre>"
+			  		+ "* modelId : It is the combined value of 'namespace:name:version' of the model<br/>"
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java [hunk 3]
```diff
@@ -60,7 +60,14 @@
...
+			  		+ "	Example: com.mycompany:MagneticSensor:1.0.0<br/>"
+			  		+ "* file : This is the file to be attached to the above model."
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java [hunk 4]
```diff
@@ -98,7 +105,13 @@ public AttachResult attach(
     }
   }
 
-  @ApiOperation(value = "Get the list of file attachments for a model")
+  @ApiOperation(value = "Get the list of file attachments for a model", 
+		  notes = "This method is used to get 'ALL' files attached to the specific modelId. Only 1 input is required."
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java [hunk 5]
```diff
@@ -98,7 +105,13 @@ public AttachResult attach(
...
+			  		+ "<br/>"
+			  		+ "<pre>"
+			  		+ "* modelId : It is the combined value of 'namespace:name:version' of the model<br/>"
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java [hunk 6]
```diff
@@ -117,11 +130,19 @@ public AttachResult attach(
     }
   }
 
-  @ApiOperation(value = "Get a specific file attachment for a model")
+  @ApiOperation(value = "Get a specific file attachment for a model", 
+		  notes = "This method is used to get the specified file attached to the specific modelId. It requires two inputs for proper response"
+		  		+ "<br/>"
+		  		+ "<pre>"
+		  		+ "* modelId : It is the combined value of 'namespace:name:version' of the model<br/>"
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/AttachmentController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java [hunk 7]
```diff
@@ -69,7 +72,14 @@
   @Autowired
   private IGeneratorService generatorService;
 
-  @ApiOperation(value = "Generate code for a specified platform, and extract specified path")
+  @ApiOperation(value = "Generate code for a specified platform, and extract specified path",
+		  notes = "This method allows to get code generated for a specified platform using the 'service key' value along with the 'modelId'."
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java [hunk 8]
```diff
@@ -144,7 +154,14 @@ private String getExtractPath(final HttpServletRequest request) {
     return apm.extractPathWithinPattern(bestMatchPattern, path);
   }
 
-  @ApiOperation(value = "Generate code for a specified platform")
+  @ApiOperation(value = "Generate code for a specified platform",
+		  notes = "This method allows to get code generated for a specified platform using the 'service key' value along with the 'modelId'."
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java [hunk 9]
```diff
@@ -187,11 +204,13 @@ private void writeToResponse(final HttpServletResponse response, GeneratedOutput
     return requestParams;
   }
 
-  @ApiOperation(value = "Returns all currently registered Code Generator")
+  @ApiOperation(value = "Returns all currently registered Code Generator",
+		  notes = "This method call allows to get all registered code genertors currently in the repository."
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java [hunk 10]
```diff
@@ -232,7 +251,8 @@ private boolean contains(String[] tags, String tag) {
     return false;
   }
 
-  @ApiOperation(value = "Returns a specific generator info")
+  @ApiOperation(value = "Returns a specific generator info", 
+		  notes = "This method allows to get information of a specific generator. Here the input that needs to be passed is the 'servicekey' of the generator.")
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/GeneratorController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/ModelController.java [hunk 11]
```diff
@@ -85,11 +89,13 @@ public ModelInfo getModelInfo(
 				UserContext.user(SecurityContextHolder.getContext().getAuthentication().getName()));
 	}
 	
-	@ApiOperation(value = "Returns the complete model content")
+	@ApiOperation(value = "Returns the complete model content",
+			notes = "This method call return the complete model content of the specific 'modelId' provided in the query.<br/>")
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/ModelController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/ModelController.java [hunk 12]
```diff
@@ -110,11 +116,16 @@ public ModelContent getModelContent(
 		return result;
 	}
 
-	@ApiOperation(value = "Returns the complete model content including target platform specific attributes")
+	@ApiOperation(value = "Returns the complete model content including target platform specific attributes",
+			notes = "This method call help to get the complete content of the model and its attributes defined "
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/ModelController.java]: updated

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/ModelController.java [hunk 13]
```diff
@@ -164,14 +175,18 @@ private boolean isMappingForModel(MappingModel p, Model model) {
 		}
 	}
 
-	@ApiOperation(value = "Downloads the model file")
+	@ApiOperation(value = "Downloads the model file",
+			notes = "This method call helps to get the model file download for the 'modelId' provided. "
```
[the remark this answers is not in the document]
Comment (vlvanchin) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/ModelController.java]: updated

**Pull request merged**